### PR TITLE
updated printLadder, added recursDFS, completed getWordLadderDFS

### PR DESCRIPTION
**Pull request by traj30**

### CODE REVIEW

File: WordLadder/src/assignment3/Main.java [hunk 1]
```diff
@@ -125,12 +139,95 @@ public static void initialize() {
...
+				curr = i + 1;
+				
+				if (curr >= end.length()){
+					curr = 0;
+					break;
+				}
+			}
+		}
+		
+		while (count != 4){
```
Comment (malekAlSukhni) [inline WordLadder/src/assignment3/Main.java]: why is it while count != 4? what is the 4?

Comment (traj30) [reply WordLadder/src/assignment3/Main.java]: just making sure that the index curr is less than the length of startTemp, the first word

Comment (malekAlSukhni) [reply WordLadder/src/assignment3/Main.java]: So why 4, this should take all cases into account, so it should be start.length() right? This way only accounts for the situation of a certain size. Same with the rest of the 4's, unless they serve a different purpose?

Comment (traj30) [reply WordLadder/src/assignment3/Main.java]: good point, it should function properly and more universally if we change all of the 4's .

**Comment by malekAlSukhni** [comment]
:shipit: looks good to me

**Pull request merged**